### PR TITLE
Add admin API to delete a pet service account for a user

### PR DESCRIPTION
**Pull request by rtitle**
No JIRA. 

@jmthibault79 @dvoet FYI

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/test/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutesSpec.scala [hunk 1]
```diff
@@ -199,7 +199,34 @@ class UserRoutesSpec extends FlatSpec with Matchers with ScalatestRouteTest {
...
   }
 
+  "DELETE /api/admin/user/{userSubjectId}/petServiceAccount" should "delete a pet service account for a user" in withAdminRoutes { (samRoutes, adminRoutes) =>
+    // create a user
+    Post("/register/user") ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.Created
+      responseAs[UserStatus] shouldEqual UserStatus(UserStatusDetails(defaultUserId, defaultUserEmail), Map("ldap" -> true, "allUsersGroup" -> true, "google" -> true))
+    }
 
+    // create a pet service account
```
Comment (jmthibault79) [inline src/test/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutesSpec.scala]: I would like one more create and delete

Comment (rtitle) [reply src/test/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutesSpec.scala]: Create and delete pet you mean? Ok I can add

**Review by jmthibault79 — APPROVED** [review]
Woot

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutes.scala [hunk 2]
```diff
@@ -87,6 +87,15 @@ trait UserRoutes extends UserInfoDirectives {
                       }
                     }
                   }
+                } ~
+                pathPrefix("petServiceAccount") {
+                  pathEndOrSingleSlash {
+                    delete {
+                      complete {
+                        userService.deleteUserPetServiceAccount(WorkbenchUserId(userId)).map(_ => StatusCodes.OK)
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutes.scala]: return no content status code

File: src/main/resources/swagger/api-docs.yaml [hunk 3]
```diff
@@ -75,6 +75,19 @@ paths:
...
+      summary: deletes a user's pet service account
+      parameters:
+        - in: path
+          description: User ID whose pet to delete
+          name: userId
+          required: true
+          type: string
+      operationId: deletePet
+      tags:
+        - Admin
```
Comment (dvoet) [inline src/main/resources/swagger/api-docs.yaml:98]: responses section

**Review by dvoet — APPROVED** [review]
just a couple minor things then 👍

**Pull request merged**